### PR TITLE
[WIP] Adding GcStatusMonitor to log Major GCs

### PR DESCRIPTION
**Pull request by kelvinfann**
# [Note: this diff will not build until [pr](https://github.com/prestodb/airlift/pull/2 ) is landed and presto has updated its dependency]

## Summary
In this diff we add GcStatusMonitor to have richer GC logs.

### Key Changes
* Added a class GcStatusMonitor to log full GCs with active task level information
* Hooked into the main server module to initiate on boot
* Added tests to ensure major GCs do not break the server.

## Testing Notes:
Because its rather difficult to deterministically force a GC while active tasks I simply ran the integration tests repeatedly and recorded the output to make sure it worked successfully

## Risk Notes:
This PR is relatively low risk:
- The server boot is tested via the unit test
- The receiving of GC notifications and the logic under it was tested manually
  - Even on the off chance something had an issue, the standard java8 library already has try/catches [the listener](https://www.programcreek.com/java-api-examples/?code=lambdalab-mirror/jdk8u-jdk/jdk8u-jdk-master/src/share/classes/sun/management/NotificationEmitterSupport.java)

```
2019-10-03T21:14:14.543-0500 INFO Major GC: application 47817ms, stopped 536ms:: 2.18GB -> 1.12GB
2019-10-03T21:14:14.554-0500 INFO | Query ID                    | Memory Pool ID | Max Bytes | Free Bytes | Reserved Bytes |
2019-10-03T21:14:14.554-0500 INFO | 20191004_021413_00375_utnwq | general        | 833827636 | 833747448  | 80188          |
2019-10-03T21:14:14.555-0500 INFO | Task ID                         | Query ID                    | State    | Created Ts                    | Cumulative Mem | Input Bytes | Output Bytes | Input Row Count | Output Row Count |
2019-10-03T21:14:14.555-0500 INFO | 20191004_021413_00375_utnwq.1.1 | 20191004_021413_00375_utnwq | RUNNING  | 2019-10-03T21:14:13.735-05:00 | 365331.48      | 0           | 0            | 0               | 0                |
2019-10-03T21:14:14.555-0500 INFO | 20191004_021413_00375_utnwq.3.0 | 20191004_021413_00375_utnwq | FINISHED | 2019-10-03T21:14:13.732-05:00 | 0.00           | 128         | 0            | 320             | 10               |
2019-10-03T21:14:14.555-0500 INFO | 20191004_021413_00375_utnwq.0.0 | 20191004_021413_00375_utnwq | RUNNING  | 2019-10-03T21:14:13.735-05:00 | 0.00           | 0           | 0            | 0               | 0                |
2019-10-03T21:14:14.555-0500 INFO | 20191004_021413_00375_utnwq.2.1 | 20191004_021413_00375_utnwq | FINISHED | 2019-10-03T21:14:13.733-05:00 | 0.00           | 0           | 67608        | 7512            | 7512             |
```

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    public GcStatusMonitor(
+            SqlTaskManager taskManager)
+    {
+        this.sqlTaskManager = taskManager;
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        for (GarbageCollectorMXBean mbean : ManagementFactory.getGarbageCollectorMXBeans()) {
```
[the remark this answers is not in the document]
Comment (kelvinfann) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: after

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            }
+        }
+    }
+
+    private void onMajorGc(GarbageCollectionNotificationInfo info)
+    {
+        long applicationRuntime = max(0, info.getStartTime() - lastGcEndTime);
+        lastGcEndTime = info.getEndTime();
+        log.info(String.format(
+                "%-40s %20s %15s %20s %20s %20s %20s %20s %20s",
```
[the remark this answers is not in the document]
Comment (kelvinfann) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: I created a formater as requested

**Comment by kelvinfann** [comment]
> Nice! The slight duplication of code from JmxGcMonitor wasn't quite what I expected. I was expecting to add a listener interface to JmxGcMonitor and then register this with it.

I was under the impression that our decision was to simply listen directly to the JVM's notifications rather than create a listener ontop of an existing listener. I can certainly make a listener hook into JmxGcMonitor if we want to revisit that decision.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                "State",
+                "Created Ts",
+                "Cumulative Mem",
+                "Input Size",
+                "Output Size",
+                "Input Positions",
+                "Output Positions"));
+
+        try {
+            for (SqlTask task : sqlTaskManager.getAllTasks()) {
```
[the remark this answers is not in the document]
Comment (kelvinfann) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: I ended up getting the query contexts from the tasks inorder to filter out the queries that don't actually have active tasks open on the node. lmk if you prefer it done differently

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                "Max Bytes",
+                "Free Bytes",
+                "Reserved Bytes"));
+
+        for (QueryContext ctx : queryContexts) {
+            MemoryPoolInfo memoryPoolInfo = ctx.getMemoryPool().getInfo();
+            queryLogList.add(Arrays.asList(
+                    ctx.getQueryId().toString(),
+                    ctx.getMemoryPool().getId().toString(),
+                    Long.toString(memoryPoolInfo.getMaxBytes()),
```
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: There is interesting point in time memory reservation information in QueryContext.queryMemoryContext which is specific to the query. There are two versions in there, not sure what the difference is? Aggregated and local? @arhimondr 

I would swap the two memory pool values you have here for point in time values from those.

File: presto-main/src/test/java/com/facebook/presto/server/TestServer.java [hunk 5]
```diff
@@ -122,6 +124,37 @@ public void testServerStarts()
         assertEquals(response.getStatusCode(), OK.getStatusCode());
     }
 
+    @Test(timeOut = 10000)
+    public void testServerStartsWithGC()
+            throws Exception
+    {
+        // force a GC
+        long gcCount = getNumGcs();
```
Comment (aweisberg) [inline presto-main/src/test/java/com/facebook/presto/server/TestServer.java]: Once committer review time comes I think you will have to make this more black box (look up memory limits and then allocate to trigger full GC) or remove the test.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        try {
+            onMajorGcLogic();
+        }
+        catch (Throwable throwable) {
+            log.error(throwable);
+        }
+    }
+
+    private void onMajorGcLogic()
```
Comment (aweisberg) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: onMajorGCLogging?

I suspect this name will trigger someone to ask for a change.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                "Created Ts",
+                "User Memory",
+                "System Memory",
+                "Input Bytes",
+                "Output Bytes",
+                "Input Row Count",
+                "Output Row Count"));
+
+        for (SqlTask task : sqlTasks) {
+            TaskInfo taskInfo = task.getTaskInfo();
```
[the remark this answers is not in the document]
Comment (kelvinfann) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: should be groupped now

File: presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java [hunk 8]
```diff
@@ -246,6 +246,16 @@ public synchronized MemoryPool getMemoryPool()
         return memoryPool;
     }
 
+    public long getMaxUserMemory()
+    {
+        return this.maxUserMemory;
```
Comment (aweisberg) [inline presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java]: Don't specify this when you don't need it.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 9]
```diff
@@ -195,8 +194,16 @@ private void logTaskInfos(List<SqlTask> sqlTasks)
...
-        for (String row : StringTableUtils.getTableStrings(taskLogInfoList)) {
-            log.info(row);
+        logInfoTable(taskLogInfoList);
+    }
+
+    private void logTaskInfosByQueryID(Map<QueryId, List<SqlTask>> queryIDToSqlTaskMap)
+    {
+        for (Map.Entry<QueryId, List<SqlTask>> mapEntry : queryIDToSqlTaskMap.entrySet()) {
+            // Log the Query
+            logQueryInfo(sqlTaskManager.getQueryContext(mapEntry.getKey()));
```
Comment (aweisberg) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Sorry I think I changed my mind (only a little) on this. Run through this loop twice. First time just log the query info. Second time just log the task infos. Each task info contains the query ID so someone can look at the queries and then look at tasks from individual queries.

The tasks still need to be grouped by query so this isn't wasted work.

There could be O(100) queries. You need to be able to just see just the queries first and then separately do a text search for the query id to find the tasks. That's what the example I gave showed.

Comment (aweisberg) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Also one more pony... can you sort the queries before you log them in descending order by max total memory?

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 10]
```diff
@@ -165,7 +165,6 @@ private void logTaskInfos(List<SqlTask> sqlTasks)
         ArrayList<List<String>> taskLogInfoList = new ArrayList<>();
         taskLogInfoList.add(Arrays.asList(
                 "Task ID",
-                "Query ID",
```
Comment (aweisberg) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: I think you will end up needing this if we list all the queries first without interleaving them with their tasks. Sorry! I am learning more about what we need as I review this.

File: presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java [hunk 11]
```diff
@@ -246,6 +246,16 @@ public synchronized MemoryPool getMemoryPool()
...
+    }
+
+    public long getMaxTotalMemory()
+    {
+        return this.maxTotalMemory;
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/memory/QueryContext.java]: ditto about `this.`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class GcStatusMonitor
+{
+    private final Logger log = Logger.get(GcStatusMonitor.class);
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: `private static final`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+    private final NotificationListener notificationListener = (notification, ignored) -> onNotification(notification);
+    private final SqlTaskManager sqlTaskManager;
+
+    @Inject
+    public GcStatusMonitor(
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Inject `MBeanServer` here. It is available and bound with the Guice context

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+            SqlTaskManager taskManager)
+    {
+        this.sqlTaskManager = taskManager;
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: `requireNonNull`

Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: nit: please be consistent with naming. Use either `taskManager` or `sqlTaskManager`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+
+public class GcStatusMonitor
+{
+    private final Logger log = Logger.get(GcStatusMonitor.class);
+    private final NotificationListener notificationListener = (notification, ignored) -> onNotification(notification);
+    private final SqlTaskManager sqlTaskManager;
+
+    @Inject
+    public GcStatusMonitor(
+            SqlTaskManager taskManager)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: nit: do not wrap

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+                ManagementFactory.getPlatformMBeanServer().removeNotificationListener(objectName, notificationListener);
+            }
+            catch (JMException ignored) {
+            }
+        }
+    }
+
+    private synchronized void onNotification(Notification notification)
+    {
+        if ("com.sun.management.gc.notification".equals(notification.getType())) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Please extract the `"com.sun.management.gc.notification"` into a static constant with a meaningful name, e.g.: `GC_NOTIFICATION_TYPE`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+            ObjectName objectName = mbean.getObjectName();
+            try {
+                ManagementFactory.getPlatformMBeanServer().removeNotificationListener(objectName, notificationListener);
+            }
+            catch (JMException ignored) {
+            }
+        }
+    }
+
+    private synchronized void onNotification(Notification notification)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: This method doesn't alter any state. Why does it have to be `synchronized`?

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        try {
+            onMajorGcLogging();
+        }
+        catch (Throwable throwable) {
+            log.error(throwable);
+        }
+    }
+
+    private void onMajorGcLogging()
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: `logActiveTasks`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        // We only care about active tasks
+        List<SqlTask> activeSqlTasks = getActiveSqlTasks();
+        Map<QueryId, List<SqlTask>> activeQueriesToTasksMap = getQueriesToTaskMap(activeSqlTasks);
+        logTaskInfosByQueryID(activeQueriesToTasksMap);
+    }
+
+    private Map<QueryId, List<SqlTask>> getQueriesToTaskMap(List<SqlTask> sqlTasks)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Use the java stream API inline + `toImmutableListMultimap` collector instead

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+                queriesToActiveTasks.put(queryContext.getQueryId(), new ArrayList<>());
+            }
+
+            queriesToActiveTasks.get(queryContext.getQueryId()).add(task);
+        }
+
+        return queriesToActiveTasks;
+    }
+
+    private List<SqlTask> getActiveSqlTasks()
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Use the java stream API inline + `toImmutableList` collector instead

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+                    Long.toString(taskIOStats.getInputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputDataSize().getTotalCount()),
+                    Long.toString(taskIOStats.getInputPositions().getTotalCount()),
+                    Long.toString(taskIOStats.getOutputPositions().getTotalCount())));
+        }
+
+        logInfoTable(taskLogInfoList);
+    }
+
+    private void logTaskInfosByQueryID(Map<QueryId, List<SqlTask>> queryIDToSqlTaskMap)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: nit: How about `logFormatted`

Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: In the Presto projects (as in many Java projects) we use reverse method ordering. If method `A` is called by method `B`, the method ordering will be `B` -> `A`. Please reorder.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        for (Map.Entry<QueryId, List<SqlTask>> mapEntry : queryIDToSqlTaskMap.entrySet()) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: static import `Entry`

Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Also you can use streaming API.

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+    }
+
+    private void logInfoTable(List<List<String>> table)
+    {
+        for (String row : StringTableUtils.getTableStrings(table)) {
+            log.info(row);
+        }
+    }
+
+    private void logQueryInfo(QueryContext ctx)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: how about `logFormatted`?

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        logInfoTable(Arrays.asList(
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: We prefer `ImmutableList.of` over the `Arrays.asList`

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 25]
```diff
@@ -0,0 +1,206 @@
...
+                        "Max Total Bytes"),
+                Arrays.asList(
+                        ctx.getQueryId().toString(),
+                        Long.toString(ctx.getMaxUserMemory()),
+                        Long.toString(ctx.getMaxTotalMemory()))));
+    }
+
+    private void logTaskInfos(List<SqlTask> sqlTasks)
+    {
+        ArrayList<List<String>> taskLogInfoList = new ArrayList<>();
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: Please use streaming API + toImmutableList collector

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java [hunk 26]
```diff
@@ -0,0 +1,206 @@
...
+                sqlTasks.add(task);
+            }
+        }
+
+        return sqlTasks;
+    }
+
+    private void logInfoTable(List<List<String>> table)
+    {
+        for (String row : StringTableUtils.getTableStrings(table)) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: static import `getTableStrings`

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+package com.facebook.presto.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Math.max;
+
+public class StringTableUtils
+{
+    public static String getShortestTableStringFormat(List<List<String>> table)
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: We don't define unchecked exception in the method interface. Please remove the `IllegalArgumentException`.

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        if (table.size() == 0) {
+            return "";
+        }
+
+        int tableWidth = table.get(0).size();
+        int[] lengthTracker = new int[tableWidth];
+        for (List<String> row : table) {
+            if (row.size() != lengthTracker.length) {
+                throw new IllegalArgumentException("all lists in table must be the same size");
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: format(`all rows in the table are expected to have exactly same number of columns: %s != %s`, tableWidth, row.size())

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+            throws IllegalArgumentException
+    {
+        if (table.size() == 0) {
+            return "";
+        }
+
+        int tableWidth = table.get(0).size();
+        int[] lengthTracker = new int[tableWidth];
+        for (List<String> row : table) {
+            if (row.size() != lengthTracker.length) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: `s/lengthTracker.length/tableWidth/`

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+                lengthTracker[i] = max(row.get(i).length(), lengthTracker[i]);
+            }
+        }
+
+        String formatString = "|";
+        for (int maxLen : lengthTracker) {
+            formatString += " %-" + maxLen + "s |";
+        }
+
+        return formatString.trim();
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: Why the `trim` is needed?

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 31]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public static List<String> getTableStrings(List<List<String>> table)
+            throws IllegalArgumentException
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: Ditto about exception

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 32]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        String formatString = getShortestTableStringFormat(table);
+        ArrayList<String> outputList = new ArrayList<String>(table.size());
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: streaming API + toImmutableList collector

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 33]
```diff
@@ -0,0 +1,63 @@
...
+        for (int i = 0; i < table.size(); i++) {
+            outputList.add(String.format(formatString, table.get(i).toArray()));
+        }
+
+        return outputList;
+    }
+
+    private StringTableUtils() {}
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: nit: please move it to the top

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 34]
```diff
@@ -0,0 +1,63 @@
...
+
+import static java.lang.Math.max;
+
+public class StringTableUtils
+{
+    public static String getShortestTableStringFormat(List<List<String>> table)
+            throws IllegalArgumentException
+    {
+        if (table.size() == 0) {
+            return "";
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: Maybe throw the `IllegalArgumentException` instead

File: presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java [hunk 35]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+        int tableWidth = table.get(0).size();
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/util/StringTableUtils.java]: Maybe throw the `IllegalArgumentException` if the table width is 0

File: presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java (hunk 15 above)
Comment (aweisberg) [reply presto-main/src/main/java/com/facebook/presto/util/GcStatusMonitor.java]: What do you mean don't wrap? I am trying to fix this up and it seems like the style matches other places I looked at.

**Comment by arhimondr** [comment]
Superseded by https://github.com/prestodb/presto/pull/13774

**Pull request closed without merging**